Streamline documentation to reflect project "piper" (#875)
SAP/jenkins-library: documentation/docs/guidedtour.md

Cx Server is a life-cycle management tool to bootstrap a pre-configured Jenkins instance within minutes.
All required plugins and shared libraries are included automatically.
It is based on Docker images provided by project "piper".

To get started, initialize Cx Server using this `docker run` command:

```sh
docker run -it --rm -u $(id -u):$(id -g) -v "${PWD}":/cx-server/mount/ ppiper/cx-server-companion:latest init-cx-server
```

This will create a few files in your current working directory.
The shell script `cx-server` and the configuration file `server.cfg` are of special interest.

Now you can start the Jenkins server by using the following command:

```sh
chmod +x ./cx-server
./cx-server start
```

For more information on Cx Server and how to customize your Jenkins, have a look at the [Operations Guide for Cx Server][devops-docker-images-cxs-guide].

### On your own: Custom Jenkins Setup

If you use your own Jenkins installation you need to care for "Piper" specific configuration.
This option should only be considered if you know why you need it, otherwise using Cx Server life-cycle management will make your life much easier.

> Follow [my own Jenkins installation][guidedtour-my-own-jenkins] for some hints, if you chose to go this path.

Reviewer: Do we really want to advertise that here. Especially, we should state that there is no support for that or sth. similar.

Reviewer 2: I think the "piper on own Jenkins" is a valid use case. We should keep this.

Author: I did keep it. I think in most cases it will cause more work then using Cx Server, but I think the new description makes that clear. I hope this is a description that is ok for all of us.

* You have installed a Linux system with at least 4 GB memory. **Note:** We have tested our samples on Ubuntu 16.04. On Microsoft Windows, you might face some issues.
* You have installed the newest version of Docker. See [Docker Community Edition](https://docs.docker.com/install/). **Note:** we have tested on Docker 18.09.6.
* Your system has access to [GitHub.com][github].

> ## Recommended: Install the Cx Server Life-cycle Management for Jenkins

Reviewer 3: ```suggestion
## **Recommended:** Install the Cx Server Life-cycle Management for Jenkins
```

Cx Server is a life-cycle management tool to bootstrap a pre-configured Jenkins instance within minutes.
All required plugins and shared libraries are included automatically.

> It is based on Docker images provided by project "piper".

Reviewer 3: ```suggestion
It is based on Docker images provided by project "Piper".
```

Author: I guess we have many places where we wrote it with a small "p".. Is there an official spelling?

Reviewer 3: Yes, there is: https://www.sapbrandtools.com/naming-center/#/requests/7058

> To get started, initialize Cx Server using this `docker run` command:

Reviewer 3: ```suggestion
To get started, initialize Cx Server by using this `docker run` command:
```

```sh
docker run -it --rm -u $(id -u):$(id -g) -v "${PWD}":/cx-server/mount/ ppiper/cx-server-companion:latest init-cx-server
```

> This will create a few files in your current working directory.

Reviewer 3: ```suggestion
This creates a few files in your current working directory.
```

The shell script `cx-server` and the configuration file `server.cfg` are of special interest.

> Now you can start the Jenkins server by using the following command:

Reviewer 3: ```suggestion
Now, you can start the Jenkins server by using the following command:
```

```sh
chmod +x ./cx-server
./cx-server start
```

> For more information on Cx Server and how to customize your Jenkins, have a look at the [Operations Guide for Cx Server][devops-docker-images-cxs-guide].

Reviewer 3: ```suggestion
For more information on the Cx Server and how to customize your Jenkins, have a look at the [Operations Guide for Cx Server][devops-docker-images-cxs-guide].
```

### On your own: Custom Jenkins Setup

> If you use your own Jenkins installation you need to care for "Piper" specific configuration.

Reviewer 3: ```suggestion
If you use your own Jenkins installation, you need to care for the configuration that is specific to project "Piper".
```

> This option should only be considered if you know why you need it, otherwise using Cx Server life-cycle management will make your life much easier.

Reviewer 3: ```suggestion
This option should only be considered if you know why you need it, otherwise using the Cx Server life-cycle management makes your life much easier.
```

> Follow [my own Jenkins installation][guidedtour-my-own-jenkins] for some hints, if you chose to go this path.

Reviewer 3: ```suggestion
If you choose to go this path, follow [my own Jenkins installation][guidedtour-my-own-jenkins] for some hints.
```

## (Optional) Sample Application

> !!! info "Choosing the best sample application"

Reviewer 3: ```suggestion
**Info:**
```

Author: @OliverNocon @marcusholl Do we have some sort of agreement for such things? My understanding was that the "boxes" created with this `!!! info` thing is what we want as they look more pleasant in the generated documentation.

Reviewer 2: Agreements for such things? Not that I know ...

Since @SarahNoack is deeply involved in the docu business also elsewhere I think she has the best understanding in how doing things like this (e.g. deciding when to use boxes).
